src/EquationTypes.jl

```diff
@@ -0,0 +1,25 @@
+"""
+Types for dispatching on different equation sets for KiD
+"""
+
+export AbstractStyle
+export AbstractMoistureStyle
+export AbstractPrecipitationStyle
+
+export EquilibriumMoisture
+export NonEquilibriumMoisture
+
+export NoPrecipitation
+export Precipitation0M
+export Precipitation1M
+
+abstract type AbstractStyle end
+abstract type AbstractMoistureStyle <: AbstractStyle end
+abstract type AbstractPrecipitationStyle <: AbstractStyle end
+
+struct EquilibriumMoisture <: AbstractMoistureStyle end
```

Comment:
Let's also add: "NoMoisture" as a moisture type, and it can be an argument to the init_condition fn

Reply:
Done. We should work on adding all the function specifications for those different types. Right now KiD will only work with `EquilibriumMoisture` and `NoPrecipitation`

Should be done in next PRs. This one is already big